docs/concepts.md

```diff
@@ -7,15 +7,20 @@ toc: false
 summary: "Here we introduce some basic concepts & give a broad technical overview of Hudi"
 ---
 
-Apache Hudi (pronounced “Hudi”) provides the following primitives over datasets on DFS
+Apache Hudi (pronounced “Hudi”) provides the following streaming primitives over datasets on DFS
 
  * Upsert                     (how do I change the dataset?)
- * Incremental consumption    (how do I fetch data that changed?)
+ * Incremental pull           (how do I fetch data that changed?)
 
+In this section, we will discuss key concepts & terminologies that are important to understand, to be able to effectively use these primitives.
 
-In order to achieve this, Hudi maintains a `timeline` of all activity performed on the dataset, that helps provide `instantaenous` views of the dataset,
-while also efficiently supporting retrieval of data in the order of arrival into the dataset.
-Such key activities include
+## Timeline
+At its core, Hudi maintains a `timeline` of all actions performed on the dataset at different `instants` of time that helps provide instantaenous views of the dataset,
```

Comment:
'At its core, Hudi maintains a `timeline` of all actions performed on the dataset at different `instants` of time.This helps provide instantaneous views of the dataset`.
Note the spelling of instantaneous as well.